test/widgets/cards_test.dart

```diff
@@ -0,0 +1,344 @@
+    expect(find.byIcon(Icons.access_time_outlined), findsOneWidget);
+
+    //start location is shown
+    expect(find.byKey(const Key('start')), findsOneWidget);
+
+    //end location is shown
+    expect(find.byKey(const Key('end')), findsOneWidget);
+  });
+
+  group('PendingRideCard', () {
+    ride = RideFactory().generateFake(
+      status: RideStatus.pending,
+      rider: NullableParameter(ProfileFactory().generateFake(id: 1)),
+      seats: 1,
+    );
+    drive = DriveFactory().generateFake(rides: [ride], seats: 2);
+
+    setUp(() {
+      whenRequest(processor).thenReturnJson([]);
+    });
+
+    testWidgets('shows the correct information', (WidgetTester tester) async {
+      await pumpMaterial(tester, PendingRideCard(ride, reloadPage: () {}, drive: drive));
+
+      //Profile Widget from rider is shown
+      final Finder profileWidgetFinder = find.byType(ProfileWidget);
+      expect(profileWidgetFinder, findsOneWidget);
+      final ProfileWidget profileWidget = tester.widget(profileWidgetFinder);
+      expect(profileWidget.profile, ride.rider);
+
+      //Extra time is shown
+      expect(find.byKey(const Key('extraTime')), findsOneWidget);
+
+      //Seats indicator is shown
+      final Finder seatsIndicator = find.byType(IconWidget);
+      expect(seatsIndicator, findsOneWidget);
+      expect(find.descendant(of: seatsIndicator, matching: find.byIcon(Icons.chair)), findsOneWidget);
+
+      //Price is shown
+      expect(find.byKey(const Key('price')), findsOneWidget);
+
+      //Approve button is shown
+      expect(find.byKey(const Key('approveButton')), findsOneWidget);
+
+      //Reject button is shown
+      expect(find.byKey(const Key('rejectButton')), findsOneWidget);
+    });
+
+    group('approve Ride', () {
+      Future<void> openApproveDialog(WidgetTester tester) async {
+        //need scaffold for dialog
+        await pumpMaterial(tester, Scaffold(body: PendingRideCard(ride, reloadPage: () {}, drive: drive)));
+        //open dialog
+        final Finder approveButton = find.byKey(const Key('approveButton'));
+        await tester.tap(approveButton);
+        //load dialog
+        await tester.pumpAndSettle();
+      }
+
+      testWidgets('can approve Ride if possible', (WidgetTester tester) async {
+        await openApproveDialog(tester);
+
+        //confirm dialog
+        final Finder confimrButton = find.byKey(const Key('approveConfirmButton'));
+        expect(confimrButton, findsOneWidget);
+        await tester.tap(confimrButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byKey(const Key('approveSuccesSnackbar')), findsOneWidget);
+        verifyRequest(
+          processor,
+          urlMatcher: equals('/rest/v1/rpc/approve_ride'),
+          methodMatcher: equals('POST'),
+          bodyMatcher: equals({'ride_id': ride.id}),
+        ).called(1);
+      });
+
+      testWidgets('can not approve Ride if not possible', (WidgetTester tester) async {
```

Comment:
Maybe include something along the lines of "too many seats" instead of "not possible" for more clarity when just reading the error messages

Reply:
did that